File: examples/telephone/second/kid.rb
require 'dramatis/actor'

require 'mangler'

class Kid

  include Dramatis::Actor
  
  def initialize name, next_kid = nil
    @name = name
    @next = next_kid
  end

  def to_s
    @name
  end

  def whisper what
    mangle( @next.nil? && @next.whisper( mangle( what )  ) || what )
  end

  def mangle what
    Mangler.mangle what
  end

end

File: examples/telephone/second/kid.py
import dramatis

from  mangler import Mangler

class Kid( dramatis.Actor ):
  
  def __init__(self, name, next_kid = None):
      super(Kid,self).__init__()
      self._name = name
      self._next = next_kid

  def __str__(self):
      return self._name
  
  def whisper(self, what):
      return self.mangle( self._next and self._next.whisper( what ) or what )

  def mangle(self, what):
      return Mangler.mangle( what )
